cmake_modules/cmake_uninstall.cmake.in

```diff
@@ -1,21 +1,22 @@
-IF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
-  MESSAGE(FATAL_ERROR "Cannot find install manifest: \"@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt\"")
-ENDIF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
+if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt")
+  message(FATAL_ERROR "Cannot find install manifest: \"${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt\"")
+else()
```

Comment:
Thank you for your feedback. You’re right about the message(FATAL_ERROR ...) command causing an immediate exit, simplifying the script by avoiding the else block. 

I appreciate your guidance on this! I oversaw this, my bad.

Another minor change: I think we can use lowercase options for some commands. The CMake documentation often uses lowercase for commands and keywords. Since this script was written in 2010, I suppose there was a change in what is considered "better practice", but I haven't checked syntax changes over time, it's a supposition.  Let me know if we should adopt lowercase syntax for this file or keep uppercase. 

The final version with new changes would be like this:

```cmake
if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt")
  message(FATAL_ERROR "Cannot find install manifest: \"${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt\"")
endif()

file(READ "${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt" files)
string(REGEX REPLACE "\n" ";" files "${files}")

foreach(file IN LISTS files)
  message(STATUS "Uninstalling \"${ENV{DESTDIR}}${file}\"")
  if(EXISTS "${ENV{DESTDIR}}${file}")
    execute_process(
      command "${CMAKE_COMMAND}" -E remove "${ENV{DESTDIR}}${file}"
      output_variable rm_out
      result_variable rm_retval
    )
    if(NOT "${rm_retval}" STREQUAL "0")
      message(FATAL_ERROR "Problem when removing \"${ENV{DESTDIR}}${file}\"")
    endif()
  else()
    message(STATUS "File \"${ENV{DESTDIR}}${file}\" does not exist.")
  endif()
endforeach()
```

Practical question: I can revert the last commit if that's good enough. In this case, using a hard reset to undo a commit in a GitHub PR is possible and keeps a cleaner history. If you prefer, I can just use git revert as a new commit. Let me know if there is a guideline about that.

Reply:
```cmake
# Check if install manifest file exists
if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt")
  message(FATAL_ERROR "Cannot find install manifest: \"${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt\"")
endif()

# Read install manifest file
file(READ "${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt" files)

# Replace newline with semicolons
string(REGEX REPLACE "\n" ";" files "${files}")

# Iterate over  list:
foreach(file IN LISTS files)
  message(STATUS "Uninstalling \"${ENV{DESTDIR}}${file}\"")
  # Check if file exists
  if(EXISTS "${ENV{DESTDIR}}${file}")
    # Remove file
    execute_process(
      COMMAND "${CMAKE_COMMAND}" -E remove "${ENV{DESTDIR}}${file}"
      OUTPUT_VARIABLE rm_out
      RESULT_VARIABLE rm_retval
    )
    # Check if the removal was successful
    if(NOT "${rm_retval}" STREQUAL "0")
      message(FATAL_ERROR "Problem when removing \"${ENV{DESTDIR}}${file}\"")
    endif()
  else()
    # File does not exist
    message(STATUS "File \"${ENV{DESTDIR}}${file}\" does not exist.")
  endif()
endforeach()

```

Reply:
Added comments for documentation and minor edits. It's a revert with some added edits.

PS: I used `git reset --hard HEAD~1` and `push --force`. It seems it's not a problem in simple PRs on GitHub. Hope it's fine.